desc/optimize/_constraint_wrappers.py

```diff
@@ -189,18 +209,34 @@ def update_constraint_target(self, eq_new):
 
         # There is probably a more clever way of doing this, but for now we just
         # remove fixed parameters from A and b again by the same loop as in factorize
+        # Actually A (unscaled linear constraint matrix without any degenerate rows)
+        # does not change here, but still recompute it while updating others
         A, b, xp, unfixed_idx, fixed_idx = remove_fixed_parameters(A, b, xp)
 
         # compute x_scale
-        x_scale = self._objective.x(*self._objective.things)
-        Dnew = jnp.where(jnp.abs(x_scale) < 1e2, 1, jnp.abs(x_scale))
+        if x_scale == "auto":
```

Comment:
ohhh i tihnk i understood you before. I thought you meant `x_scale` argument to `optimize` which is internal to the optimizer, so shouldn't affect any of this stuff? I think you can get rid of this